### ISSUE TITLE
Rule request: AvoidLeakingRetryEventConsumers

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#254]
We found a memory leak in certain use of resilience4j Retry. 
```java
import io.github.resilience4j.retry.Retry;

public class Foo {
    Retry retryField;
    RetryRegistry reg;
    final AtomicInteger retryCountField = new AtomicInteger();

    Foo() {
        retryField = reg.retry("one per service");
        retryField.getEventPublisher().onRetry(event -> retryCountField.getAndIncrement()); // good
    }

    void callService() {
        AtomicInteger retryCountLocal = new AtomicInteger();
        retryField.getEventPublisher().onRetry(event -> retryCountLocal.getAndIncrement()); // bad, lambda and AtomicInt leak

        Retry retryLocal = reg.retry("one per method call");
        retryLocal.getEventPublisher().onRetry(event -> retryCountLocal.getAndIncrement()); // good, no leak

        // same for onSuccess, onError, onIgnoredError
    }
}
```
The lambda is the Event Consumer and it will be added to the publisher. There is no way to unregister. 
Besides a memory leak, it will also take more and more CPU time, to update all those consumers.
